ui/pages/confirm-transaction/confirm-transaction.component.js

```diff
@@ -109,30 +104,14 @@ const ConfirmTransaction = () => {
   const prevParamsTransactionId = usePrevious(paramsTransactionId);
   const prevTransactionId = usePrevious(transactionId);
 
-  const _beforeUnload = useCallback(() => {
-    setIsMounted(false);
-
-    if (pollingToken) {
-      disconnectGasFeeEstimatePoller(pollingToken);
-      removePollingTokenFromAppState(pollingToken);
-    }
-  }, [pollingToken]);
+  usePolling({
+    startPollingByNetworkClientId: gasFeeStartPollingByNetworkClientId,
+    stopPollingByPollingToken: gasFeeStopPollingByPollingToken,
+    networkClientId: 'mainnet',
```

Comment:
can we avoid hardcoding this? Passing down the networkClientId from the req object? Or falling back to the globally selected networkClientId if not present?

Reply:
https://github.com/MetaMask/metamask-extension/pull/22770/commits/16a39fe1914c21df7fe29780e3b8ec3b3c3c19e4